fits.py

```diff
@@ -0,0 +1,96 @@
+import os
+from pathlib import Path
+import numpy as np
+import astropy.io.fits as fits
+from astropy.table import Table
+import argparse
+
+
+NEW_COMMENT = (
+    ' ',
+    '              THE AXAF GUIDE and ACQUISITION STAR CATALOG V1.7',
+    ' ',
+    '               An all-sky astrometric and photometric catalog',
+    '                   prepared for the operation of AXAF.',
+    ' ',
+    'This file contains data for one of the 9537 regions constituting',
+    'the AXAF Guide and Acquisition Star Catalog (AGASC) version 1.7',
+    'Additional information on the AGASC may be obtained from the',
+    'AXAF Science Center, Smithsonian Astrophysical Observatory,',
+    '60 Garden St., Cambridge, MA 02138, or in tables elsewhere',
+    'in this set of volumes (see the "comments" file).',
+    'Null values are listed as -9999.',
+    ' ',
+    'V1.7 changes values for MAG_ACA, MAG_ACA_ERR, RSV1, RSV2 and RSV3',
+    ' ',
+    ' ',
+    'File written on Sept 5 2018',
+    ' ')
+
+
+def get_options():
+    parser = argparse.ArgumentParser(
+       description="Make new AGASC fits files from old AGASC and new h5 source")
+    parser.add_argument("--h5", default='/proj/sot/ska/data/agasc/agasc1p7.h5')
+    parser.add_argument("--olddir", default="/data/agasc1p6/agasc")
+    parser.add_argument("--out", default="./testagasc")
+    args = parser.parse_args()
+    return args
+
+
+def update_file(src, dest, newdata):
+    print(src, dest)
+    hdu = fits.open(src)
+    hdr = hdu[0].header
+    # Update AGASC VERSION
+    hdr['EXTVER'] = 17
+
+    for row in hdu[1].data:
+        # Set RSV1 to be -9999.0 by default (for most stars this is a no-op, already -9999.0)
+        row['RSV1'] = -9999.0
+        # Find the index of the AGASC ID we have in the sorted table of newdata
+        idx = np.searchsorted(newdata['AGASC_ID'], row['AGASC_ID'])
+        if newdata['AGASC_ID'][idx] != row['AGASC_ID']:
```

Comment:
I think this is a bit belt-and-suspenders, but I can't see a way that the end result here can be wrong.